src/mixins/java/org/spongepowered/common/mixin/core/world/entity/EntityMixin.java

```diff
@@ -237,6 +236,27 @@
         this.impl$isConstructing = false;
     }
 
+    @Override
+    public boolean bridge$setPosition(final Vector3d position) {
+        if (this.removed) {
+            return false;
+        }
+
+        try (final CauseStackManager.StackFrame frame = PhaseTracker.getCauseStackManager().pushCauseFrame()) {
+            frame.pushCause(SpongeCommon.activePlugin());
+            frame.addContext(EventContextKeys.MOVEMENT_TYPE, MovementTypes.PLUGIN);
+            final Vector3d destinationPosition = this.impl$fireMoveEvent(PhaseTracker.SERVER, position);
+            if (destinationPosition == null) {
+                return false;
+            }
+            if (this instanceof ServerPlayerBridge) {
```

Comment:
Despite the desire to remove duplicated code, I have always disliked these instance checks when it feels better to practice OOP. By having it in ServerPlayer. What do you think?

Reply:
Sure - but I also want to make a policy clear here because it's not that clear at the moment, what is our policy on overriding bridge methods?

I was under the (recent) impression that overriding bridge methods in sub classes shouldn't be done, though I'm seeing plenty of examples in the codebase. Platform bridges are also inherently designed to be overridden. Am I wrong on that, did I get the wrong end of the stick?

That said, I'd just change this to be an `impl$...` method that is overridden in a subclass. That's what I did elsewhere.

Reply:
Bridge methods should only exist if we need to invoke code from non-mixins against mixed classes. Otherwise it is `impl$`/etc. As for OOP, there is nothing that prevents overriding bridge methods or even `impl$` methods. In fact, I 100% encourage it.

Reply:
Alright. I was overriding impl$ methods anyway, so that's likely what will happen here too.